docs/remote_debugging.md

```diff
@@ -0,0 +1,218 @@
+# Introduction
+
+Remote debugging with Ruby debugger means:
+
+1. Invoke your program as a remote debuggee.
+2. Connect the debuggee to supported platforms:
+    - Debugger console
+    - VSCode (or other DAP supporting clients)
+    - Chrome DevTools (or other CDP supporting clients)
```

Comment:
Maybe we should link here to the sections for VSCode and Chrome so that reader doesn't have to read the whole thing to find it.